discourse-doctor

```diff
@@ -0,0 +1,403 @@
+##
+dump_yaml() {
+  echo -e "\n\n==================== YML DUMP ====================" | tee -a $LOG_FILE
+  echo Dumping $app_name.yml | tee -a $LOG_FILE
+  sed -e '/\s*#.*$/d' -e '/^\s*$/d' -e 's/  DISCOURSE_SMTP_PASSWORD:\(.*\)/  DISCOURSE_SMTP_PASSWORD: REDACTED/g' containers/$app_name.yml  | tee -a $LOG_FILE
+  echo -e "\n\n" | tee -a $LOG_FILE
+}
+
+##
+## read a variable from the config file
+##
+read_config() {
+  config_line=`egrep "^  #?$1:" $web_file`
+  read_config_result=`echo $config_line | awk  --field-separator=":" '{print $2}'`
+  read_config_result=`echo $read_config_result | sed "s/^\([\"']\)\(.*\)\1\$/\2/g"`
+}
+
+##
+## send a test message
+##
+email_test() {
+  echo -e "\n\n==================== MAIL TEST ===================="
+  echo "I recommend that you get a test address from http://www.mail-tester.com/ , but you "
+  echo "can just send a test message to yourself."
+  EMAIL=`echo $DEVELOPER_EMAILS |cut -d , -f 1`
+  read -p "Email address for mail test? ('n' to skip) [$EMAIL]: " new_value
+  if [ ! -z "$new_value" ]
+  then
+    EMAIL="$new_value"
+  fi
+  if [ "$new_value" != "n" ] && [ "$new_value" != "N" ]
+  then
+    echo "Sending mail to $EMAIL. . . "| tee -a $LOG_FILE
+    sudo docker exec -w /var/www/discourse -i $app_name rake emails:test[$EMAIL] |& tee -a $LOG_FILE
+  else
+    echo "Mail test skipped."
+  fi
+}
+
+##
+## Get container name
+##
+
+docker_check() {
+  if [ -f /var/discourse/containers/app.yml ]
+  then
+    echo "YML=app.yml"
+    app_name=app
+    data_name=app
+    web_file=containers/$app_name.yml
+    data_file=containers/$app_name.yml
+  fi
+  if [ -f /var/discourse/containers/web_only.yml ]
+  then
+    echo "YML=web_only.yml"
+    app_name=web_only
+    web_file=containers/$app_name.yml
+  fi
+  if [ -f /var/discourse/containers/data.yml ]
+  then
+    data_name=data
+    data_file=containers/$data_name.yml
+  fi
+
+  echo "==================== docker ps ====================" | tee -a $LOG_FILE
+  docker ps |& tee -a $LOG_FILE
+
+  echo "==================== end docker ps ====================" | tee -a $LOG_FILE
+  docker ps| tail -n +2 > /tmp/$UUID-docker.txt
+  if grep $app_name /tmp/$UUID-docker.txt
+  then
+    echo "Discourse container $app_name is running"
+  else
+    echo "==================== SERIOUS PROBLEM!!!! ===================="
+    echo "$app_name not running!"
+    echo "Attempting to rebuild"
+    echo "==================== REBUILD LOG ===================="
+    ./launcher rebuild $app_name |& tee -a $LOG_FILE
+    echo "==================== END REBUILD LOG ===================="
+    docker ps| tail -n +2 > /tmp/$UUID-docker.txt
+    if grep $app_name /tmp/$UUID-docker.txt
+    then
+      echo "Discourse container $app_name is running!"
+    else
+      echo "$app_name still not running!"
+      echo "You should probably remove any non-standard plugins and rebuild."
+      NO_CONTAINER='y'
+    fi
+  fi
+}
+
+##
+## Discourse version
+##
+get_discourse_version () {
```

Comment:
there's `get_discourse_version()` and `discourse_version()` could we make the difference more explicit please?

Reply:
This should be cleaned up. Sorry.

Reply:
Yeah. Those aren't great names. How about  `get_discourse_version` that `wget`s `$VERSION_HOSTNAME` and `compare_discourse_versions` that checks whether `localhost` and `$DISCOURSE_HOSTNAME` match?

I'll remove the comments too.

Also, `check_IP_match` (also pulled from `./discourse-setup`) wasn't getting called. I tweaked those comments and added a call to `check_IP_match`.